kani-driver/build.rs

```diff
@@ -3,11 +3,28 @@
 
 use std::env::var;
 
+use cargo_metadata::MetadataCommand;
+
 fn main() {
     // We want to know what target triple we were built with, but this isn't normally provided to us.
     // Note the difference between:
     // https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-crates
     // https://doc.rust-lang.org/cargo/reference/environment-variables.html#environment-variables-cargo-sets-for-build-scripts
     // So "repeat" the info from build script (here) to our crate's build environment.
     println!("cargo:rustc-env=TARGET={}", var("TARGET").unwrap());
+
+    // Force this script to re-run if `kani-verifier/Cargo.toml` changed.
+    // Otherwise, we may not get the up-to-date version in `kani-driver`.
+    println!("cargo:rerun-if-changed=../Cargo.toml");
+
+    // Pass the `kani-verifier` version through the build environment.
+    println!("cargo:rustc-env=KANI_VERIFIER_VERSION={}", kani_verifier_version());
+}
+
+/// Retrieve the `kani-verifier` version using `cargo metadata`
+fn kani_verifier_version() -> String {
```

Comment:
Shouldn't the `kani-driver` version always match `kani-verifier`? I don't understand why we are doing this.

Reply:
I've opened #2626 to ensure these are the same versions. We just use the CARGO_PKG_VERSION now.